### ISSUE TITLE
[YYC] Modifying a struct inline as returned by a function crashes

### ISSUE THREAD

**Issue by thennothinghappened** [YoYoGames/GameMaker-Bugs#4509]
### Description
On YYC, if you modify a struct returned from a function in the same line you call that function it will crash with a cryptic error:
```
ERROR!!! :: ############################################################################################
ERROR in
action number 1
of Create Event
for object <undefined>:


I32 argument is unset
############################################################################################
gml_GlobalScript_test_script (line 7)
```

 I don't actually know if VM's behaviour here would be how I would expect it personally, but I found this after adding the [Input](https://github.com/offalynne/Input) library to my project, which uses this behaviour extensively.

Tested on **Beta IDE v2024.200.0.471 Beta Runtime v2024.200.0.487**

### Steps To Reproduce
Here's the test code I used in the submitted project:
```gml
function static_struct() {
	static struct = {};
	return struct;
}

static_struct().blah = 2;
```
This works on VM, but on YYC will crash with the error above.

[3bbc049b-2de7-4b24-a341-066e4579a58d](https://api.gamemaker.io/api/github/downloads/3bbc049b-2de7-4b24-a341-066e4579a58d)

**Comment by thennothinghappened** [comment]
Don't know how I managed to miss this, but it seems more broadly that accessors in general are broken in YYC.

The following code from my game causes an identical crash:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/35243139/e16fde1f-a597-4917-abf6-11819f74e71a)

**Comment by thennothinghappened** [comment]
After some more testing, it looks like the `.` accessor isn't working with certain keywords, and in certain cases:
```gml
self.arr = [];

self[$ "arr"][0] = 0;

// Crashes
self.arr[0] = 0;
```

Also interestingly, the errors given by YYC and VM differ for this silly line:
```gml
undefined.blah = 2;
```

VM:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/35243139/8ef37f07-cd60-400a-a55e-41174e0a02c7)

YYC:
![image](https://github.com/YoYoGames/GameMaker-Bugs/assets/35243139/b15e59c3-6af0-4d36-bdc9-f4228eb03a21)
...which reminds me a bit of "I32 argument is unset", but now it's undefined.

**Comment by thennothinghappened** [comment]
Here's the offending generated line.
```cpp
YY_STACKTRACE_LINE(6);
oW__This__0BBF685F8_arr=1;
FREE_RValue( &__ret1__ );
YYGML_Variable_SetValue( __tmp3__, kVARID_self_arr, (int)0, &oW__This__0BBF685F8_arr );
;
```
`__tmp3__` is unset. Going off one single mention of this function [here](https://yal.cc/gamemaker-on-global-variables/) and with zero actual knowledge to its inner workings, nor its function signature, I set this value to 0 as it was in that example, and the value was correctly set, so that's nice (though I'll again reiterate I have _no_ idea what I just did)

**Comment by rwkay** [comment]
Fixed in 2024.2 - it was due to the fix for https://github.com/YoYoGames/GameMaker-Bugs/issues/4375, updated that fix to offset this issue, apologies.

**Comment by mgeddesGM** [comment]
verified as of  IDE v2024.200.0.476  Runtime v2024.200.0.491
